dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/RuleActionAssignValueImplementer.java

```diff
@@ -43,12 +43,14 @@
 import org.hisp.dhis.rules.models.RuleActionAssign;
 import org.hisp.dhis.rules.models.RuleEffect;
 import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
 
 /**
  * @author Zubair Asghar
  */
 @Slf4j
 @Component( "org.hisp.dhis.programrule.engine.RuleActionAssignValueImplementer" )
+@Transactional
```

Comment:
We shouldn't use Transactional at the class level and we should use it with `(read-only = true)` if there are no writing happening.

Reply:
resolved